app-modules/service-management/src/Jobs/ServiceMonitoringCheckJob.php

```diff
@@ -79,24 +80,41 @@ public function uniqueFor(): int
 
     public function handle(): void
     {
-        $response = Http::maxRedirects(15)
-            ->get($this->serviceMonitoringTarget->domain);
+        try {
+            $response = Http::maxRedirects(15)
+                ->get($this->serviceMonitoringTarget->domain);
+
+            $history = $this->serviceMonitoringTarget->histories()->create([
+                'response' => $response->status(),
+                'response_time' => $response->transferStats->getTransferTime() ?? 0,
+                'succeeded' => $response->status() === 200,
+            ]);
+
+            if ($response->status() !== 200) {
+                /** @var Collection<int, User> $recipients */
+                $recipients = $this->serviceMonitoringTarget->users()->get();
+
+                $this->serviceMonitoringTarget->teams()->each(function ($team) use ($recipients) {
+                    /** @var Collection<int, User> $users */
+                    $users = $team->users()->get();
+
+                    $recipients->concat($users)->unique();
+                });
+
+                Notification::send($recipients, new ServiceMonitoringNotification($history));
+            }
+        } catch (ConnectionException $e) {
+            $history = $this->serviceMonitoringTarget->histories()->create([
```

Comment:
Just before this, let's also `report()` the Exception if the message doesn't contain `Could not resolve host`. That way we can be notified if this fails for some other strange reason we have not yet accounted for.